File: guess_user_number/java/guess_user_number.java
import java.util.Scanner;

public class guess_user_number {
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);

		int min=1,max=101;
		boolean found=false;

		while (!found) {
			int mid = (max-min)/2+min;
			System.out.print(mid + "? >>> ");
			String response = scan.nextLine().toUpperCase();
			if (response.equals("Y")) {
				System.out.println("Found it: "+mid);
				found = true;
			} else if (response.equals("L")) {
				min = mid;
			} else if (response.equals("H")) {
				max = mid;
			}
		}
	}
}

File: guess_user_number/python/guess_user_number.py
#!/usr/bin/python

min=1
max=100

print "Think of a number from "+str(min)+" to "+str(max)+".  Then, tell the computer if its guess is correct (Y), lower than the number (L), or higher than the number (H)."

found=False

while not found:
	mid = int((max-min)/2+min)
	response = raw_input(str(mid)+"? >>> ")
	response = response.upper()
	if response == "Y":
		print "found it!!"
		found=True
	elif response == "L":
		min=mid
	elif response == "H":
		max=mid
